scripts/zones/AlTaieu/mobs/Qnxzomit.lua

```diff
@@ -3,20 +3,23 @@
 --  Mob: Qn'xzomit
 -- Note: Pet for JOJ
 -----------------------------------
-mixins = { require("scripts/mixins/job_special") }
 local ID = require("scripts/zones/AlTaieu/IDs")
 -----------------------------------
 local entity = {}
 
 entity.onMobSpawn = function(mob)
-    mob:timer(math.random(7500, 10000),
-        function(mobArg)
-            mobArg:useMobAbility(xi.jsa.MIJIN_GAKURE)
-            mobArg:timer((2000),
-            function(mobArg2)
-                mobArg2:setHP(0)
+    -- Sets a timer to self destruct betwen 25 - 35 seconds after being damaged
+    mob:addListener("TAKE_DAMAGE", "QNXZOMIT_JOL_TAKE_DAMAGE", function(xzomit, amount)
+        if amount > 0 and xzomit:getLocalVar("control") == 0 then
+            mob:setLocalVar("control", 1)
+            xzomit:timer((30 + math.random(-5, 5)) * 1000, function(xzomitArg)
+                xzomitArg:useMobAbility(731) -- Mijin Gakure
+                xzomitArg:timer(1000, function(xzomitTimer)
+                    xzomitArg:setHP(0)
+                end)
```

Comment:
Is this correct? I thought JoJ pets just exploded after a set amount of time, very similar to Bomb Queen.

Reply:
Wiki mentions that they blow soon after they're attacked, and that jailer can have up to 5 spawned. With this system, the xzomit will just fall dead before the next one is even spawned. Essentially only rendering the jailer to only ever have 1 at a time.
What we could instead do is set a drastically varying timer in which the xzomit kerplode. Anywhere between 10 secpnd and 2 minutes or something.
Moreover we can set them to only explode if engaged and not roaming.